multicluster/service-mirror/cluster_watcher.go

```diff
@@ -682,6 +698,16 @@ func (rcsw *RemoteClusterServiceWatcher) getMirrorServices() ([]*corev1.Service,
 	if err != nil {
 		return nil, err
 	}
+	if len(services) == 0 {
+		serviceList, err := rcsw.localAPIClient.Client.CoreV1().Services("").List(context.Background(), metav1.ListOptions{LabelSelector: labels.SelectorFromSet(matchLabels).String()})
```

Comment:
Here we first look for services in the cache, and if none are present, we fall back to calling this API?  I know we do this kind of thing elsewhere but it freaks me out a bit.  For example, the cache may have services but still be out of date.  I think that if we want to guarantee up-to-date data, we should just call the API without trying the cache first.  If being up-to-date is less important, we can just use the cache.

Reply:
Yea this was an unfortunate change that was needed only for unit tests; I couldn't reproduce the informer not being updated locally running unit tests, or with an actual cluster.

I'll change this to just always hit the API because we need it to be up to date for the tests.